File: push/src/main/java/com/nihility/utils/PrivilegeElevator.java
package com.nihility.utils;

import com.topjohnwu.superuser.Shell;
import com.xiaomi.xmsf.BuildConfig;

public class PrivilegeElevator {

    static {
        // Set settings before the main shell can be created
        Shell.enableVerboseLogging = BuildConfig.DEBUG;
        Shell.setDefaultBuilder(Shell.Builder.create()
                .setFlags(Shell.FLAG_REDIRECT_STDERR)
                .setTimeout(10)
        );
    }

    public static void tryToElevate() {
        Shell.getShell();
    }
}

File: push/src/main/java/com/nihility/utils/PrivilegeElevator.kt
package com.nihility.utils

import com.topjohnwu.superuser.Shell
import com.xiaomi.xmsf.BuildConfig

class PrivilegeElevator {
    companion object {
        init {
            Shell.enableVerboseLogging = BuildConfig.DEBUG
            Shell.setDefaultBuilder(
                Shell.Builder.create()
                    .setFlags(Shell.FLAG_REDIRECT_STDERR)
                    .setTimeout(10)
            )
        }

        @JvmStatic
        fun tryToElevate() {
            Shell.getShell()
        }
    }
}
